app/src/main/java/com/a0xffffffff/dinesum/RequestInfoActivity.java

```diff
@@ -0,0 +1,215 @@
+                    JSONObject jsonObject = new JSONObject(serverResponse);
+                    JSONArray jsonArray = jsonObject.getJSONObject("result").getJSONArray("photos");
+                    String photoReference = jsonArray.getJSONObject(0).getString("photo_reference");
+
+                    String photoReferenceUrl = "https://maps.googleapis.com/maps/api/place/photo?maxwidth=400&photoreference=" + photoReference + "&key=" + getResources().getString(R.string.google_places_api_key);;
+                    stringBuilder.append(photoReferenceUrl);
+                    handler.postDelayed(runnable, 0);
+                } catch (Exception ex) {
+                    ex.printStackTrace();
+                }
+            }
+        }).start();
+
+        restaurantName.setText(restaurantNameIntent);
+        restaurantAddress.setText(restaurantAddressIntent);
+        restaurantNumber.setText(restaurantNumberIntent);
+        requestPrice.setText(requestPriceIntent);
+        partyName.setText(requestNameIntent);
+        partySize.setText(partySizeIntent);
+        requestTime.setText(requestTimeIntent);
+
+        final boolean isRequester = requesterId.equals(Profile.getCurrentProfile().getId());
+        switch (requestStatusIntent) {
+            case RequestState.PENDING:
+                requestViewingState = isRequester ?
+                        RequestViewingState.REQUESTER_PENDING :
+                        RequestViewingState.RESERVER_PENDING;
+                break;
+            case RequestState.CLAIMED:
+                requestViewingState = isRequester ?
+                        RequestViewingState.REQUESTER_CLAIMED :
+                        RequestViewingState.RESERVER_CLAIMED;
+                break;
+            case RequestState.COMPLETED:
+                requestViewingState = isRequester ?
+                        RequestViewingState.REQUESTER_COMPLETED :
+                        RequestViewingState.RESERVER_COMPLETED;
+                break;
+            case RequestState.PAID:
+                requestViewingState = isRequester ?
+                        RequestViewingState.REQUESTER_PAID :
+                        RequestViewingState.RESERVER_PAID;
+                break;
+        }
+
+        initializeButtons();
+        final String profilePicUrl = "https://graph.facebook.com/" + requesterId + "/picture?type=square";
+        Picasso.with(getApplicationContext()).load(profilePicUrl).into(requesterProfilePicture);
+    }
+
+    private void initializeButtons() {
+        switch (requestViewingState) {
+            case RequestViewingState.REQUESTER_PENDING:
+                button2.setText(getResources().getString(R.string.remove_request));
+                button3.setVisibility(View.GONE);
+                break;
+            case RequestViewingState.REQUESTER_CLAIMED:
+                button2.setText(getResources().getString(R.string.mark_as_pending));

```

Comment:
Is this so that if someone claims your request, you can essentially force them to unclaim it and mark the request as pending again so someone else can claim it?

Reply:
Yes, do we not have that in our current design? I think this way is nice